File: arkhe_multilang/rosetta/Agent.scala
class Agent(val name: String, var counter: Int = 0)

def sendMessage(sender: Agent, receiver: Agent): Unit = {
    receiver.counter += 1
    println(s"${sender.name} -> ${receiver.name}: ${receiver.counter}")
}

object Main extends App {
    val alice = new Agent("Alice")
    val bob = new Agent("Bob")
    sendMessage(alice, bob)
    sendMessage(alice, bob)
}

File: arkhe_multilang/rosetta/Agent.swift
import Foundation

class Agent {
    let name: String
    var counter: Int = 0
    init(name: String) {
        self.name = name
    }
}

func sendMessage(sender: Agent, receiver: Agent) {
    receiver.counter += 1
    print("\(sender.name) -> \(receiver.name): \(receiver.counter)")
}

let alice = Agent(name: "Alice")
let bob = Agent(name: "Bob")
sendMessage(sender: alice, receiver: bob)
sendMessage(sender: alice, receiver: bob)
